src/core/simulation.jl

```diff
@@ -110,6 +110,61 @@ function unregister_recorders!(internal::SimulationInternal)
     end
 end
 
+function _add_initial_condition_caches(
+    sim::SimulationInternal,
+    stage::Stage,
+    caches::Union{Nothing, Vector{<:AbstractCache}},
+)
+    initial_conditions = stage.internal.psi_container.initial_conditions
+    for (ic_key, init_conds) in initial_conditions.data
+        _create_cache(ic_key, caches)
+    end
+    return
+end
+
+function _create_cache(ic_key::ICKey, caches::Union{Nothing, Vector{<:AbstractCache}})
+    return
+end
+
+function _create_cache(
+    ic_key::ICKey{TimeDurationON, T},
+    caches::Union{Nothing, Vector{<:AbstractCache}},
+) where {T <: PSY.Device}
+
+    cache_keys = CacheKey.(caches)
+    if isempty(cache_keys) || !in(CacheKey(TimeStatusChange, T), cache_keys)
+        cache = TimeStatusChange(T, PSI.ON)
+        push!(caches, cache)
+    end
+    return
+end
+
+function _create_cache(
+    ic_key::ICKey{TimeDurationOFF, T},
+    caches::Union{Nothing, Vector{<:AbstractCache}},
+) where {T <: PSY.Device}
+
+    cache_keys = CacheKey.(caches)
+    if isempty(cache_keys) || !in(CacheKey(TimeStatusChange, T), cache_keys)
+        cache = TimeStatusChange(T, PSI.ON)
+        push!(caches, cache)
+    end
+    return
+end
+
+function _create_cache(
+    ic_key::ICKey{EnergyLevel, T},
+    caches::Union{Nothing, Vector{<:AbstractCache}},
+) where {T <: PSY.Device}
+
```

Comment:
I don't see why the caches will take a nothing here

Reply:
I should actually remove that, but the reason I initial added it there was this function returned `Nothing`
https://github.com/NREL-SIIP/PowerSimulations.jl/blob/6ffdfa826540d66837af12af5b011d3c12e1ee77/src/core/simulation.jl#L242-L254